resources/wikia/wikia.wikibits.js

```diff
@@ -109,54 +110,145 @@ var importArticle = (function() {
 			|| $.inArray( 'bureaucrat', window.wgUserGroups ) > -1 ) ),
-		// TODO: i18n
 		missingText = {
-			single: '%1s was not found (requested by user-supplied javascript)',
-			multiple: '%1s %2s were not found (requested by user-supplied javascript)'
+			single:  'import-article-missing-single',
+			multiple: 'import-article-missing-multiple'
 		},
-		// TODO: i18n
 		moreText = {
-			single: '(and one more article)',
-			multiple: '(and %d more articles)'
+			single: 'import-article-missing-more-single',
+			multiple: 'import-article-missing-more-multiple'
+		},
+		notJsText = {
+			single:  'import-article-not-js-single',
+			multiple: 'import-article-not-js-multiple'
 		};
 
-	return function( missing ) {
+	function showBannerNotification(articles, baseText) {
 		var missingLength;
 
 		// Don't show notificaton for regular users
-		if ( !reportMissing ) {
+		if (!reportMissing) {
 			return;
 		}
 
-		if ( !$.isArray( missing ) ) {
-			missing = [ missing ];
+		if (!$.isArray(articles)) {
+			articles = [articles];
 		}
 
 		// Use BannerNotification to show the error to the user
-		if (window.BannerNotification && (missingLength = missing.length)) {
+		if (window.BannerNotification && (missingLength = articles.length)) {
 			var moreLength = missingLength - 1,
-				message = missingText[ missingLength < 2 ? 'single' : 'multiple' ],
-				more = moreText[ moreLength < 2 ? 'single' : 'multiple' ];
+				baseMessageName = baseText[ missingLength < 2 ? 'single' : 'multiple' ],
+				moreMessageName = moreText[ moreLength < 2 ? 'single' : 'multiple'],
+				message;
+
+			message = mw.message(baseMessageName).params([
+				'"' + articles[0] + '"',
+				mw.message(moreMessageName).params([moreLength]).escaped()
+			]).escaped();
+
+			$(function () {
+				new window.BannerNotification(message, 'error').show();
+			});
+		}
+	}
+
+	function importArticleMissing(missing) {
+		showBannerNotification(missing, missingText);
+	}
 
-			message = message
-				.replace( '%1s', '"' + missing[ 0 ] + '"' )
-				.replace( '%2s', more.replace( '%d', moreLength ) );
+	function importNotJsFailed(missing) {
+		showBannerNotification(missing, notJsText);
+	}
 
-			new window.BannerNotification(message, 'error').show();
+	return {
+		importArticleMissing: importArticleMissing,
+		importNotJsFailed: importNotJsFailed
+	};
+}());
+
+/**
+ * Imports script from provided JS page name
+ * Page has to be in MediaWiki namespace and has .js extension
+ *
+ * Article name can point a page on other wikia (e.g. 'external:otherwikiasubdomain:Pagename.js')
+ *
+ * @param {array} articles Names of pages to import without namespace prefix
+ */
+var importWikiaScriptPages = (function () {
+	var namespacePrefix = 'MediaWiki:',
+		externalPrefix = 'external:',
+		externalPrefixLength = externalPrefix.length;
+
+	function importWikiaScriptPages(articles) {
+		var articlesToImport = [],
+			articlesFailed = [];
+
+		for (var i = 0; i < articles.length; i++) {
```

Comment:
I think we should check if ```articles``` is an array. What do you think about allowing importing one script with parameter as a string?